File: Programming EF Code First/chapter6/Chapter6/BAGA_FluentAPI/DataAccess/BreakAwayContext.cs
using System.Data.Entity;
using Model;
using System.Data.Common;

namespace DataAccess
{
  public class BreakAwayContext : DbContext
  {
    public BreakAwayContext()
    { }

    public BreakAwayContext(string databaseName)
      : base(databaseName)
    { }

    public BreakAwayContext(DbConnection connection)
      : base(connection, contextOwnsConnection: false)
    { }

    public DbSet<Destination> Destinations { get; set; }
    public DbSet<Lodging> Lodgings { get; set; }
    public DbSet<Trip> Trips { get; set; }
    public DbSet<Person> People { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder)
    {
      // Entity Type Configuration
      modelBuilder.Configurations.Add(new DestinationConfiguration());
      modelBuilder.Configurations.Add(new LodgingConfiguration());
      modelBuilder.Configurations.Add(new TripConfiguration());
      modelBuilder.Configurations.Add(new PersonConfiguration());
      modelBuilder.Configurations.Add(new InternetSpecialConfiguration());
      modelBuilder.Configurations.Add(new ActivityConfiguration());
      modelBuilder.Configurations.Add(new PersonPhotoConfiguration());
      modelBuilder.Configurations.Add(new ReservationConfiguration());
      
      // Complex Type Configuration
      modelBuilder.Configurations.Add(new AddressConfiguration());
      modelBuilder.ComplexType<PersonalInfo>();
    }
  }
}

File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/Fluent/DataAccess/BreakAwayContext.vb
﻿Imports System.Data.Entity
Imports Model
Imports System.Data.Common

Namespace DataAccess
  Public Class BreakAwayContext
	  Inherits DbContext
	Public Sub New()
	End Sub

	Public Sub New(ByVal databaseName As String)
		MyBase.New(databaseName)
	End Sub

	Public Sub New(ByVal connection As DbConnection)
		MyBase.New(connection, contextOwnsConnection:= False)
	End Sub

	Public Property Destinations() As DbSet(Of Destination)
	Public Property Lodgings() As DbSet(Of Lodging)
	Public Property Trips() As DbSet(Of Trip)
	Public Property People() As DbSet(Of Person)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
	  ' Entity Type Configuration
	  modelBuilder.Configurations.Add(New DestinationConfiguration())
	  modelBuilder.Configurations.Add(New LodgingConfiguration())
	  modelBuilder.Configurations.Add(New TripConfiguration())
	  modelBuilder.Configurations.Add(New PersonConfiguration())
	  modelBuilder.Configurations.Add(New InternetSpecialConfiguration())
	  modelBuilder.Configurations.Add(New ActivityConfiguration())
	  modelBuilder.Configurations.Add(New PersonPhotoConfiguration())
	  modelBuilder.Configurations.Add(New ReservationConfiguration())

	  ' Complex Type Configuration
	  modelBuilder.Configurations.Add(New AddressConfiguration())
	  modelBuilder.ComplexType(Of PersonalInfo)()
	End Sub
  End Class
End Namespace